state/core_access.go

```diff
@@ -203,6 +195,15 @@ func (ca *CoreAccessor) SubmitPayForBlob(
 		return nil, fmt.Errorf("account for signer %s not found", author)
 	}
 
+	gas := cfg.GasLimit()
+	if gas == 0 {
+		msg, err := apptypes.NewMsgPayForBlobs(author.String(), appconsts.Version, libBlobs...)
+		if err != nil {
+			return nil, fmt.Errorf("failed to create msg pay-for-blobs: %w", err)
+		}
+		gas = ca.estimateGasForBlobs(msg)
```

Comment:
it would be nice if our utility method would. just accept the author + blobs and do the msg construction there instead of inline here as this method is already super long.